### ISSUE TITLE
boto_secgroup has a bug when either vpc_name or vpc_id is not specified

### ISSUE THREAD

**Issue by bogdanr** [saltstack/salt#29186]
The state that I am applying looks like this which is very close to what the documentation specifies:

```salt
AWS-SecurityGroup:
    boto_secgroup.present:
        - name: cloudhero
        - description: My security group
        - rules:
            - ip_protocol: tcp
              from_port: 80
              to_port: 80
              cidr_ip:
                - 10.0.0.0/8
                - 192.168.0.0/16
        - region: us-east-1
        - keyid: <keyid>
        - key: <key>
```

The result after running this state is:

```salt
CloudHero:
----------
          ID: AWS-SecurityGroup
    Function: boto_secgroup.present
        Name: cloudhero
      Result: False
     Comment: Security group cloudhero created. cloudhero security group configuration could not be retrieved. cloudhero security group configuration could not be retrieved.
     Started: 10:41:04.362445
    Duration: 285.972 ms
     Changes:   
              ----------
              new:
                  ----------
                  secgroup:
                      None
              old:
                  ----------
                  secgroup:
                      None

Summary for CloudHero
------------
Succeeded: 0 (changed=1)
Failed:    1
------------
Total states run:     1
Total run time: 285.972 ms
```

I looks to me there might be two issues here:

1. The security group will get created but rules are not added.
2. The configuration could not be retrieved

Since the security group gets created in my AWS account but without any of the rules I can only speculate the VPC name or ID shouldn't be required. 

Nevertheless, most people will have only one VPC anyways and having to specify the vpc mandatory makes things quite inflexible.

I tried this on the latest version of 2015.8 from git also.

**Comment by jfindlay** [comment]
@bogdanr, thanks for the report.  I redacted your `keyid` and `key` fields since I was unsure you had already done so.

**Comment by andrei821** [comment]
I also have this problem and working around it is quite difficult.

**Comment by rallytime** [comment]
@bogdanr Can you elaborate on the behavior you're seeing vs. what you're expecting? I'm only a little bit familiar with this state module, so I am having a difficult time figuring out what bug I should be looking for, or what you're expecting to happen when you don't specify either a vpc_name or vpc_id.

**Comment by rallytime** [comment]
I am also wondering if you're seeing something similar as reported/discussed in #29078?

**Comment by bogdanr** [comment]
This was fixed by Ruxandra will this pull request:
https://github.com/saltstack/salt/pull/29558

**Comment by rallytime** [comment]
@bogdanr Good eye! We'll get this closed up once the fix is merged. Thank you!

**Comment by rallytime** [comment]
Closing since fix was merged.
